File: backend/tests/test_frontend_api.js
const http = require('http');

async function testApi() {
  // Test login
  console.log("Testing login endpoint...");
  
  const postData = JSON.stringify({
    email: "admin@test.com",
    password: "Test123456"
  });
  
  const options = {
    hostname: 'backend',
    port: 8000,
    path: '/api/v1/auth/login',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Content-Length': Buffer.byteLength(postData)
    }
  };
  
  const req = http.request(options, (res) => {
    let data = '';
    
    res.on('data', (chunk) => {
      data += chunk;
    });
    
    res.on('end', () => {
      console.log(`Status: ${res.statusCode}`);
      console.log(`Response: ${data}`);
      
      if (res.statusCode === 200) {
        const response = JSON.parse(data);
        const token = response.access_token;
        console.log(`Token: ${token.substring(0, 50)}...`);
        
        // Test modules endpoint
        console.log("\nTesting modules endpoint...");
        
        const modulesOptions = {
          hostname: 'backend',
          port: 8000,
          path: '/api/v1/modules',
          method: 'GET',
          headers: {
            'Authorization': `Bearer ${token}`
          }
        };
        
        const modulesReq = http.request(modulesOptions, (modulesRes) => {
          let modulesData = '';
          
          modulesRes.on('data', (chunk) => {
            modulesData += chunk;
          });
          
          modulesRes.on('end', () => {
            console.log(`Status: ${modulesRes.statusCode}`);
            const modulesResponse = JSON.parse(modulesData);
            console.log(`Total modules: ${modulesResponse.total}`);
          });
        });
        
        modulesReq.on('error', (e) => {
          console.error(`Modules request error: ${e.message}`);
        });
        
        modulesReq.end();
      } else {
        console.error(`Login failed: ${data}`);
      }
    });
  });
  
  req.on('error', (e) => {
    console.error(`Request error: ${e.message}`);
  });
  
  req.write(postData);
  req.end();
}

testApi();

File: backend/tests/test_frontend_api.py
"""Test API connectivity from frontend container."""
import sys
sys.path.insert(0, "/app")

import asyncio
import httpx

async def test_api():
    """Test API connectivity."""
    async with httpx.AsyncClient() as client:
        # Test login
        print("Testing login endpoint...")
        resp = await client.post("http://backend:8000/api/v1/auth/login", json={
            "email": "admin@test.com",
            "password": "Test123456"
        })
        
        print(f"Status: {resp.status_code}")
        print(f"Response: {resp.json()}")
        
        if resp.status_code == 200:
            token = resp.json()["access_token"]
            print(f"Token: {token[:50]}...")
            
            # Test modules endpoint
            print("\nTesting modules endpoint...")
            headers = {"Authorization": f"Bearer {token}"}
            resp = await client.get("http://backend:8000/api/v1/modules", headers=headers)
            
            print(f"Status: {resp.status_code}")
            modules_data = resp.json()
            print(f"Total modules: {modules_data['total']}")
        else:
            print(f"Login failed: {resp.json()}")

if __name__ == "__main__":
    asyncio.run(test_api())
